k8s/manifests/ingress.yaml

```diff
@@ -0,0 +1,35 @@
+# This manifest is responsible for:
+# - defining the ingress rules of our application (defining paths resolution to the correct containers)
+# - defining the rules for the nginx loadbalancer and reverse proxy
+# - defining the ingress-side rules of the ClusterIssuer
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: mull-ingress
+  namespace: mull
+  annotations:
+    kubernetes.io/ingress.class: 'nginx'
+    cert-manager.io/cluster-issuer: 'letsencrypt'
+    nginx.ingress.kubernetes.io/force-ssl-redirect: 'true'
+    nginx.ingress.kubernetes.io/proxy-body-size: '10M'
+spec:
+  tls:
+    - hosts:
+        - www.mullapp.com
+      secretName: tls
+  rules:
+    - host: www.mullapp.com
+      http:
+        paths:
+          - path: /
+            backend:
+              serviceName: mull-frontend-service
+              servicePort: 4200
+          - path: /api
+            backend:
+              serviceName: mull-backend-service
+              servicePort: 3333
+          - path: /graphql
+            backend:
+              serviceName: mull-backend-service
+              servicePort: 3333
```

Comment:
I don't think it's a good idea to expose the backend, since we don't have any form of authorization in place.

Instead what should be done is that the backendUrl should be set to the FQDN given to the backend service in the Kubernetes cluster. See https://kubernetes.io/docs/tasks/access-application-cluster/connecting-frontend-backend/

Reply:
You are absolutely right! Created #150 to implement this.